vmd-server/src/main.rs

```diff
@@ -1,38 +1,27 @@
-mod kvm_util;
+// === Module declarations ====================================================
 
-#[macro_use] extern crate rocket;
+mod server;
+mod vmd;
 
-use rocket::mtls::Certificate;
+// === External crates ========================================================
 
-#[get("/")]
-fn mutual(cert: Certificate<'_>) -> String {
-    format!("Hello! Here's what we know: [{}] {}", cert.serial(), cert.subject())
-}
+use clap::Parser;
+use std::process::exit;
+use log::error;
 
-#[get("/", rank = 2)]
-fn kvm_api_version() -> String {
-    match kvm_util::check_kvm_api_version() {
-        Ok(version) => format!("KVM API version: {}", version),
-        Err(e) => format!("Error: {:?}", e),
-    }
-}
+use crate::server::cli::Args;
+use crate::server::run::run;
 
-#[launch]
-fn rocket() -> _ {
-    // See `Rocket.toml` and `Cargo.toml` for TLS configuration.
-    // Run `./private/gen_certs.sh` to generate a CA and key pairs.
-    rocket::build().mount("/", routes![
-        kvm_api_version,
-        mutual
-    ])
+// === Implementations ========================================================
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+    run(&Args::parse()).await.unwrap_or_else(|e| {
+        error!("VIRTUAL MACHINE DAEMON HAS ENCOUNTERED AN ERROR AND MUST EXIT");
```

Comment:
my 2 cents for NOT SHOUTING HERE!!??! Error is an error without ALL CAPS.

Reply:
https://github.com/tiiuae/vmd/pull/35/commits/8ae70a2ee1464c56b7a99b1f59a5786c64c6922d